src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -575,17 +620,59 @@ namespace OpenMS
     /**
       @brief Returns the precursor spectrum of the scan pointed to by @p iterator
 
-      If there is no precursor scan the past-the-end iterator is returned.
+      If there is no (matching) precursor scan the past-the-end iterator is returned.
+      This assumes that precursors occur somewhere before the current spectrum
+      but not necessarily the first one from the last MS level (we double-check with
+      the annotated precursorList.
+      If precursor annotations are present, uses the native spectrum ID from the 
+      @em first precursor entry of the current scan
+      for comparisons -> Works for multiple precursor ranges from the same precursor scan
+      but not for multiple precursor ranges from different precursor scans.
+      If none are present, picks the first scan of a lower level.
     */
     ConstIterator getPrecursorSpectrum(ConstIterator iterator) const;
 
     /**
       @brief Returns the index of the precursor spectrum for spectrum at index @p zero_based_index
 
-      If there is no precursor scan -1 is returned.
+      If there is no precursor scan -1 is returned. Wraps @ref getPrecursorSpectrum(ConstIterator).
     */
     int getPrecursorSpectrum(int zero_based_index) const;
 
+    /**
+      @brief Returns the first product spectrum of the scan pointed to by @p iterator
+
+      A product spectrum is a spectrum of the next higher MS level that has the
+      current spectrum as precursor.
+      If there is no product scan, the past-the-end iterator is returned.
+      This assumes that product occurs somewhere after the current spectrum
+      and comes before the next scan that is of a level that is lower than
+      the current one.
+
+      Example:
+      MS1 - ix: 0
+        MS2 - ix: 1, prec: 0
+        MS2 - ix: 2, prec: 0 <-- current scan
+        MS3 - ix: 3, prec: 1
+        MS3 - ix: 4, prec: 2 <-- product scan
+        MS2 - ix: 5, prec: 0
+        MS3 - ix: 6, prec: 5
+      MS1 - ix: 7
+        ...  <-- Not searched anymore. Returns end of experiment iterator if not found until here.
```

Comment:
this will not be readable in doxygen.
Use 
```suggestion
\verbatim
      Example:
      MS1 - ix: 0
        MS2 - ix: 1, prec: 0
        MS2 - ix: 2, prec: 0 <-- current scan
        MS3 - ix: 3, prec: 1
        MS3 - ix: 4, prec: 2 <-- product scan
        MS2 - ix: 5, prec: 0
        MS3 - ix: 6, prec: 5
      MS1 - ix: 7
        ...  <-- Not searched anymore. Returns end of experiment iterator if not found until here.
\endverbatim
```